src/NHibernate/Mapping/ByCode/FakeModelExplicitDeclarationsHolder.cs

```diff
@@ -15,6 +15,8 @@ public class FakeModelExplicitDeclarationsHolder : IModelExplicitDeclarationsHol
 		private readonly IEnumerable<MemberInfo> idBags = Enumerable.Empty<MemberInfo>();
 		private readonly IEnumerable<MemberInfo> lists = Enumerable.Empty<MemberInfo>();
 		private readonly IEnumerable<MemberInfo> manyToManyRelations = Enumerable.Empty<MemberInfo>();
+		private readonly IEnumerable<MemberInfo> entityManyToManyRelations = Enumerable.Empty<MemberInfo>();
+		private readonly IEnumerable<MemberInfo> keyManyToManyRelations = Enumerable.Empty<MemberInfo>();
```

Comment:
Why does it need three lists? These lists are only used to know that a list item is many to many or a map key is many to many. As far as I can see, it is not used anywhere else.